src/node/element/path/data.rs

```diff
@@ -185,52 +185,71 @@ impl<'l> Parser<'l> {
             _ => return Ok(None),
         };
         self.reader.consume_whitespace();
-        let parameters = self.read_parameters()?.into();
         Ok(Some(match name {
-            'M' => Move(Absolute, parameters),
-            'm' => Move(Relative, parameters),
+            'M' => Move(Absolute, self.read_parameters(false)?.into()),
+            'm' => Move(Relative, self.read_parameters(false)?.into()),
 
-            'L' => Line(Absolute, parameters),
-            'l' => Line(Relative, parameters),
+            'L' => Line(Absolute, self.read_parameters(false)?.into()),
+            'l' => Line(Relative, self.read_parameters(false)?.into()),
 
-            'H' => HorizontalLine(Absolute, parameters),
-            'h' => HorizontalLine(Relative, parameters),
+            'H' => HorizontalLine(Absolute, self.read_parameters(false)?.into()),
+            'h' => HorizontalLine(Relative, self.read_parameters(false)?.into()),
 
-            'V' => VerticalLine(Absolute, parameters),
-            'v' => VerticalLine(Relative, parameters),
+            'V' => VerticalLine(Absolute, self.read_parameters(false)?.into()),
+            'v' => VerticalLine(Relative, self.read_parameters(false)?.into()),
 
-            'Q' => QuadraticCurve(Absolute, parameters),
-            'q' => QuadraticCurve(Relative, parameters),
+            'Q' => QuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            'q' => QuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'T' => SmoothQuadraticCurve(Absolute, parameters),
-            't' => SmoothQuadraticCurve(Relative, parameters),
+            'T' => SmoothQuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            't' => SmoothQuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'C' => CubicCurve(Absolute, parameters),
-            'c' => CubicCurve(Relative, parameters),
+            'C' => CubicCurve(Absolute, self.read_parameters(false)?.into()),
+            'c' => CubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'S' => SmoothCubicCurve(Absolute, parameters),
-            's' => SmoothCubicCurve(Relative, parameters),
+            'S' => SmoothCubicCurve(Absolute, self.read_parameters(false)?.into()),
+            's' => SmoothCubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'A' => EllipticalArc(Absolute, parameters),
-            'a' => EllipticalArc(Relative, parameters),
+            'A' => EllipticalArc(Absolute, self.read_parameters(true)?.into()),
+            'a' => EllipticalArc(Relative, self.read_parameters(true)?.into()),
 
             'Z' | 'z' => Close,
 
             _ => raise!(self, "found an unknown path command '{}'", name),
         }))
     }
 
-    fn read_parameters(&mut self) -> Result<Vec<Number>> {
+    fn read_parameters(&mut self, is_arc: bool) -> Result<Vec<Number>> {
```

Comment:
I would create a separate function for parsing parameters for commands with idiosyncrasies. Like `read_parameters_elliptical_arc`.

Reply:
done